File: src/main/scala/li/cil/oc/common/item/DataCard.scala
package li.cil.oc.common.item

import li.cil.oc.common.item.abstracts.SimpleItem
import net.minecraft.item.Item
import net.minecraft.item.Item.Properties
import net.minecraftforge.common.extensions.IForgeItem

class DataCard(props: Properties, val tier: Int) extends SimpleItem(props) with IForgeItem with traits.ItemTier {
  @Deprecated
  override def getDescriptionId = super.getDescriptionId + tier
}

File: src/main/scala/li/cil/oc/common/item/DataCard.java
package li.cil.oc.common.item;

import li.cil.oc.common.item.abstracts.SimpleItem;
import li.cil.oc.common.item.interfaces.ItemTier;
import net.minecraftforge.common.extensions.IForgeItem;

public class DataCard extends SimpleItem implements IForgeItem, ItemTier {
  public final int tier;

  public DataCard(Properties props, int tier) {
    super(props);
    this.tier = tier;
  }

  @Override
  public String getDescriptionId() {
    return super.getDescriptionId() + tier;
  }
}